ledger/src/bank_forks.rs

```diff
@@ -334,10 +335,17 @@ impl BankForks {
         Ok(())
     }
 
-    fn prune_non_root(&mut self, root: Slot) {
+    fn prune_non_root(&mut self, root: Slot, largest_confirmed_root: Option<Slot>) {
         let descendants = self.descendants();
-        self.banks
-            .retain(|slot, _| slot == &root || descendants[&root].contains(slot));
+        self.banks.retain(|slot, _| {
+            slot == &root
```

Comment:
is it comparing references? should it be `*slot == root` ?

Reply:
I guess it's what it's doing before.. just looks weird

Reply:
It's either deref all the slots, or ref all the roots. I don't have a strong preference, so let me know if you do.

Reply:
Huh... I guess rust actually does an auto-dereference for you to compare the value. I would prefer the de-reference because it's closer to what is actually happening.